update contrib guide (#532)
microsoft/presidio: CONTRIBUTING.md

### General contribution guidelines

> -   A Github issue suggesting the change should be opened prior to a PR.

Reviewer: just curious - why was the extra whitespace after the dash added, in this list and next. It doesn't make a difference visually in github, but it deviates from standard markdown syntax.

Author: auto indent with vscode

-   All contributions should be documented, tested and linted. Please verify that all tests and lint checks pass successfully before proposing a change.
-   To make the linting process easier, you can use [pre-commit hooks](docs/development.md#automatically-format-code-and-check-for-code-styling) to verify and automatically format code upon a git commit
-   In order for a pull request to be accepted, the CI (containing unit tests, e2e tests and linting) needs to succeed, in addition to approvals from two maintainers.
-   PRs should be small and solve/improve one issue at a time. If you have multiple suggestions for improvement, please open multiple PRs.

### How can I contribute?

-   [Testing](#how-to-test)
-   [Adding new recognizers for new PII types](#adding-new-recognizers-for-new-pii-types)

> -   [Bug fixing and general improvement](#bug-fixing-and-general-improvement)

Reviewer: ```suggestion
-   [Fixing Bugs and improving the code](#fixing-bugs-and-improving-the-code)
```

Reviewer: Just a nit to make the list language consistent

### How to test?

For Python, Presidio leverages `pytest` and `flake8`. See [this tutorial](docs/development.md#testing) on more information on testing presidio modules.

### Adding new recognizers for new PII types

Adding a new recognizer is a great way to improve Presidio. A new capability to detect a new type of PII entity improves Presidio's coverage and makes private data less accessible.

> Best practices for recognizers development [are described here](docs/analyzer/developing_recognizers.md). Please follow these guidelines when proposing new recognizers.

Reviewer: ```suggestion
Best practices for developing recognizers [are described here](docs/analyzer/developing_recognizers.md). Please follow these guidelines when proposing new recognizers.
```

> ### Bug fixing and general improvement

Reviewer: ```suggestion
### Fixing Bugs and improving the code
```
